c10/core/DispatchKeySet.h

```diff
@@ -123,7 +123,14 @@ static inline DispatchKey legacyExtractDispatchKey(DispatchKeySet s) {
   // top of existing "normal" keys like CPU/CUDA, you need to add it
   // here.  At the moment, RequiresGrad (replacement for Variable)
   // is the most likely key that will need this treatment.
-  return s.highestPriorityTypeId();
+
+  // BackendSelect is a very special dispatch key that doesn't have a dedicated backend.
+  // It was introduced only for the factory functions with TensorOptions. The logic that
+  // uses legacyExtractDispatchKey doesnt expect BackendSelect and VariableTensorId keys.
+  // VariableTensorId is being excluded from a DispatchKeySet right after dispatching
+  // (See variable_excluded_from_dispatch in TensorBody.h)
```

Comment:
that is just an example where we are checking if something was historically a variable -- this is not the place where variable is being excluded.